### ISSUE TITLE
Bug: Midi Channel Out (Midi Clone)

### ISSUE THREAD

**Issue by mheidt** [zynthian/zynthian-issue-tracking#974]
Latest Oram 64bit

When you create an "empty" MIDI chain and use Midi Out to clone the midi traffic to certain engines, it seems to work for all engines but FluidSynth. I did tests with Dexed, Obxd, SFZ Player....but FluidSynth doesn't want to receive MIDI message from a Midi Chain.

**Comment by jofemodo** [comment]
How MIDI channels are assigned? Please, tell me the details and send  the snapshot ;-)

**Comment by jofemodo** [comment]
HI @mheidt!

I just made some adjustments, including the addition of a "Active MIDI channel" flag in the admin menu.
I can't reproduce your issue after my latest changes, so probably it's solved.

Regarding the "Active MIDI channel" flag, it simplifies layering chains. If it's enabled, all chains in the active chain's MIDI channel will receive the notes, so you don't need to create a MIDI chain+routing. Of course, this is still possible and could be very interesting. Indeed you could go beyond this and create a MIDI+Audio chain, so you route MIDI to the desired chains (cloned) and route-back audio to the MIDI+Audio chain. In this way, each chain can be played separately and you have a dedicated fader for the mix of cloned chains. Cool right? The little problem is that so much routing increase latency. The simple approach using the "Active MIDI channel" flag doesn't add latency at all. Up to you.

**Comment by riban-bw** [comment]
The issue is likely to also impact zynaddsubfx. These engines use MIDI channel to differentitate between chains, with a single instance running for all chains that use it so routing the MIDI out towards one of these is likely to not work as expected. In fact routing MIDI in this way may have other issues including increased latency.

**Comment by jofemodo** [comment]
@riban, i think what you describe is a different issue, but you are right.
I tried to reproduce @mheidt  issue with latest code and i can't.

**Comment by mheidt** [comment]
I created a snapshot.
A Fluidsynth on channel 7, a Midi Chain on channel 1 but routing to FS engine.
I set my Input device into Multitimbral.
On channel 7, sound is on, on channel 1 it isn't
Furthermore the is the good case. Dexed on channel 6 and Midi chain on 2 routing to Dexed.
Works on both channels  (6 and 2)
When I downloaded the snapshot, only the engines are listed in the webconf...1 and 3, but not the midi chains.
[013-Bug974.zss.zip](https://github.com/zynthian/zynthian-issue-tracking/files/14538112/013-Bug974.zss.zip)
